src/mesh/cartesiangrid.jl

```diff
@@ -3,9 +3,9 @@
 # ------------------------------------------------------------------
 
 """
-    CartesianGrid(dims, origin, spacing)
+    CartesianGrid(dims, reference, spacing)
```

Comment:
Can we stick to the name origin to reduce the amount of changes in the docstrings and code? I believe the meaning of the variable is still the same right? You are only adding a new offset argument to the algorithms.